Assets/MRTK/Providers/WindowsMixedReality/XR2018/Controllers/WindowsMixedRealityController.cs

```diff
@@ -66,184 +66,188 @@ public WindowsMixedRealityController(TrackingState trackingState, Handedness con
 
         #region Update data functions
 
+        private static readonly ProfilerMarker UpdateControllerPerfMarker = new ProfilerMarker("[MRTK] WindowsMixedRealityController.UpdateController");
+
         /// <summary>
         /// Update the controller data from the provided platform state.
         /// </summary>
         /// <param name="interactionSourceState">The InteractionSourceState retrieved from the platform</param>
         public override void UpdateController(InteractionSourceState interactionSourceState)
         {
-            if (!Enabled) { return; }
-
-            Profiler.BeginSample("[MRTK] WindowsMixedRealityController.UpdateContoller");
+            using (UpdateControllerPerfMarker.Auto())
```

Comment:
Should this happen after the enabled check?

Reply:
Otherwise it's possible for us to get a lot of "quick" events which would make things seem fast

Reply:
that's a good question. by showing these quick-out events, we see that there is a method call adding _some_ overhead and that the stack is consistent between these and more expensive calls.

Adding the markers after the enabled check _would_ make it easier to see the impact by eliminating the 0ms events.

I'll consider and modify in a future PR. I will need to scrub the complete tree to find all instances of this type of tracing,

Reply:
https://github.com/microsoft/MixedRealityToolkit-Unity/issues/7647

Reply:
FWIW I think it would be better to take the changes now. We're already in the code and I highlighted the cases that I saw (I don't THINK there are a ton more)

Reply:
that's fair